### ISSUE TITLE
Support gradle-node-plugin

### ISSUE THREAD

**Issue by wickkidd** [renovatebot/renovate#18746]
### What would you like Renovate to be able to do?

Be able to parse the node, npm, yarn, and pnpm versions from the node block in the `build.gradle`.

the plugin is specified as

```
plugins {
  id 'com.github.node-gradle.node' version '3.2.1'
}
```

And lib versions like this https://github.com/node-gradle/gradle-node-plugin/blob/master/docs/usage.md#configuring-the-plugin

```
node {
  version = '16.14.0'
  download = true
}
```

https://github.com/node-gradle/gradle-node-plugin

### If you have any ideas on how this should be implemented, please tell us here.

There is an open issue with that plugin about being able to use files like `.nvmrc` (and I would argue `.node-version` to be non-lib-specific) to define the version which would make this a moot request, however, that may or may not happen.

Without looking at the renovate codebase, If I were trying to solve this myself I would simply use regex to look for the node block and versions within once I'd verified the existence of the plugin. Granted, that means tight coupling to said plugin which would then have to be watched and would be a pain if the interface ever changed.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by Churro** [comment]
I agree that a custom regex manager pattern would be the simplest and most effective way to keep the node version updated. If relevant for many users, it could be included as a workaround preset.

Adding this natively to Gradle manager seems the wrong spot for me considering that this plugin is not something gradle ships out-of-the-box (checkstyle, spotbugs, ...) and it serves also a highly specific use-case. Implementation-wise it would intertwine the gradle manager with the github-tags datasource and node versioning, although these two are irrelevant for all other gradle purposes.
